src/modules/uORB/uORBDeviceMaster.cpp

```diff
@@ -268,7 +268,7 @@ int uORB::DeviceMaster::addNewDeviceNodes(DeviceNodeStatisticsData **first_node,
 			max_topic_name_length = name_length;
 		}
 
-		last_node->last_pub_msg_count = last_node->node->published_message_count();
+		last_node->last_pub_msg_count = last_node->node->updates_available(0);
```

Comment:
This reads a bit weirdly. Can you add a comment explaining why you pass a 0 or perhaps add a simple getting for current generation instead?

Reply:
Added a comment, this may be the only place that needs to be done.
And also
https://github.com/PX4/Firmware/blob/a24488328f175e4657ad449996c42260e4da9c92/src/modules/uORB/Subscription.cpp#L68
```C++ 
const unsigned curr_gen = _node->updates_available(0);

// If there were any previous publications allow the subscriber to read them
_last_generation = curr_gen-math::min((unsigned)_node->get_queue_size(), curr_gen);
```
The above code can be deleted, `_last_generation` is 0 by default and there is no problem, `uORB::DeviceNode` will automatically set it to a valid value.